tests/turnstile_propagate_many.c

```diff
@@ -58,8 +62,9 @@ static void starter_routine(void *_arg) {
         assert(thread_self()->td_prio == propagator_prio(i - 1));
       }
 
-      /* propagator[i] waits for mtx[i-1] (owned by propagator[i-1]) */
-      // TODO add asserts
+      assert(td_is_locked(propagator[i]));
+      assert(td_is_locked_on_mtx(propagator[i], &mtx[i - 1]));
+      assert(mtx_owner(&mtx[i - 1]) == propagator[i - 1]);
```

Comment:
I think that `td_is_locked_on_mtx` should itself check that the thread is indeed locked and so `assert(td_is_locked(propagator[i]));` would be unnecessary.

Reply:
You're right!